### ISSUE TITLE
scale/sprt.py takes a torch dependency for a single set_num_threads call

### ISSUE THREAD

**Issue by teerthsharma** [teerthsharma/resolvent#3]
## Summary

`scale/sprt.py` implements the Wald sequential probability ratio test: thresholds
at :162, the `Sprt` class at :168, the operating-characteristic root at :206,
`expected_n` at :237, and `calibrate` at :258. The `Sprt` class itself imports
nothing beyond `math`.

The module nevertheless imports `torch` at :118, used only for
`torch.set_num_threads(2)` at :120. Any consumer of the SPRT therefore takes a
torch dependency for a thread-count call the test itself does not need.

## Effect

The SPRT is the piece of this repository most reusable outside it: win/loss
outcomes are Bernoulli, so `Sprt` plus `expected_n` answers "stop this
comparison as soon as the evidence allows, and say beforehand whether the
comparison is affordable at all". That reuse currently requires torch in
environments that have no other reason to install it, including pure-CPU and
time-budgeted ones.

## Proposed change

Move `torch.set_num_threads(2)` behind the call site that needs it, or guard the
import:

```python
try:
    import torch
    torch.set_num_threads(2)
except ImportError:
    pass
```

`scale/sprt.py:1-160`, the K1 slope-to-rate preamble, stays as is; only the
import moves.

## Limits

Whether any current caller depends on the thread cap being set at import time
rather than at first use is not established here, and the change is only safe if
none does.
